lbrynet/extras/daemon/Daemon.py

```diff
@@ -2942,6 +2944,179 @@ def jsonrpc_routing_table_get(self):
+            # to a list containing the comment IDs of the replies
+            comment_replies = {resp['id']: resp['result'] for resp in reply_lists if 'result' in resp}
+
+            # Next, we create a batch request for the actual data of all of the replies
+            # the id in this batch request is going to be in the form 'X:Y'
+            # where X is the index of the parent comment in `comments,
+            # and Y is index of the reply's ID within the list X maps to in `comment_replies`
+            full_replies_batch = [
+                rpc_body('get_comment_data', f'{parent_idx}:{idx}', comm_index=reply_id, better_keys=True)
+                for parent_idx, id_list in comment_replies.items()
+                for idx, reply_id in enumerate(id_list[0:max_replies_shown])
+            ]
+            reply_dump = await jsonrpc_batch(url, full_replies_batch)
+            del full_replies_batch
+            # This neatly orders the response into a dict to aggregate the
+            # full comments by the parent comment they're replying to
+            #
+            # WARNING: The following block is going to be saving the comment dict
+            #   objects TO `comment_replies`. This means that the lists
+            #   stored in `comments_replies` may not hold just comments, but
+            #    the ids of the comments who weren't requested due to the
+            #   maximum reply limit. They need to be either cleaned out or stored
+            #   somewhere else
+
+            for comment in reply_dump:
+                parent_index, reply_index = comment['id'].split(':')
+                parent_index, reply_index = int(parent_index), int(reply_index)
+                comment_replies[parent_index][reply_index] = comment['result']
+
+            for idx, parent_comment in enumerate(comments):
+                if 'parent_id' not in parent_comment:
+                    parent_comment['parent_id'] = None
+                parent_comment['reply_count'] = len(comment_replies[idx])
+                parent_comment['omitted'] = 0
+                if len(comment_replies[idx]) > max_replies_shown:
+                    parent_comment['omitted'] = len(comment_replies[idx]) - max_replies_shown
+
+                response['comments'].append(parent_comment)
+                response['comments'] += comment_replies[idx][0:max_replies_shown]
+            response['page_size'] = page_size if page_size is not None else -1
+            return response
+        else:
+            for id_list in reply_lists:
+                comments[id_list['id']]['reply_count'] = len(id_list['result'])
+                comments[id_list['id']]['omitted'] = len(id_list['result'])
+            response['comments'] = comments
+        del reply_lists
+        return response
+
+    @requires(WALLET_COMPONENT)
+    async def jsonrpc_comment_create(self, claim_id: str, channel_name: str,
+                                     message: str, reply_to: int = None) -> dict:
```

Comment:
Should be `channel_id` (aka channel `claim_id`) instead of `channel_name` since you might have multiple channel claims for the same channel name (very unlikely, but possible).